### PR TITLE
Avoid keyvault secret name conflict between workflow reruns

### PR DESCRIPTION
**Pull request by ouphi**
When triggering the pipeline twice, the second run fails because when deleting the resource group, the Keyvault is self deleted and the secrets stored in the Keyvault still exist. There is a conflict in the second run because the secret already exists.

This PR makes sure the secret is deleted by enabling purge_soft_delete_on_destroy config so that secret are deleted on destroy.

This PR fixed this bug: https://github.com/agera-edc/MinimumViableDataspace/issues/46

### CODE REVIEW

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -148,14 +154,31 @@ jobs:
         participant: [company1, company2]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v3
+        name: 'Publish terraform folder'
```
Comment (cpeeyush) [inline .github/workflows/cd.yaml]:
```suggestion
        name: 'Download terraform folder'
```

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -148,14 +154,31 @@ jobs:
         participant: [company1, company2]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v3
```
Comment (cpeeyush) [inline .github/workflows/cd.yaml:178]: At line 149 `continue-on-error: true` is missing which was added in previous PR #49. is that intended to remove it ?

File: .github/workflows/cd.yaml [hunk 3]
```diff
@@ -148,14 +154,31 @@ jobs:
...
+          # Authentication settings for Terraform AzureRM provider
+          # See https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs
+          ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+          ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+          ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+          ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+
+
+      - uses: actions/upload-artifact@v3
+        name: 'Remove terraform artifact'
```
Comment (cpeeyush) [inline .github/workflows/cd.yaml]: if for some reason this fails that means artifact will stay and as it contains secret information so as a follow-up story, we can use a common remote backend storage to keep this state.

Comment (cpeeyush) [reply .github/workflows/cd.yaml]: Added placeholder issue #51, we can add details later in it.

File: .github/workflows/cd.yaml [hunk 4]
```diff
@@ -148,14 +154,31 @@ jobs:
...
+        name: 'Publish terraform folder'
+        with:
+          name: terraform-${{ matrix.participant }}
+          path: deployment/terraform
 
-      - name: 'Az CLI login'
-        uses: azure/login@v1
+      - name: 'Delete terraform resources'
+        run: |
+          chmod -R u+x .terraform/providers
```
Comment (marcgs) [inline .github/workflows/cd.yaml]: Why is this required?

File: .github/workflows/cd.yaml [hunk 5]
```diff
@@ -102,8 +102,16 @@ jobs:
       - name: 'Run terraform'
         id: runterraform
         run: |
+          echo '
+          acr_resource_group = "${{ secrets.COMMON_RESOURCE_GROUP }}"
```
Comment (marcgs) [inline .github/workflows/cd.yaml]: Suggesting to indent the content of the `echo` for better readability

File: .github/workflows/cd.yaml (hunk 3 above)
Comment (marcgs) [reply .github/workflows/cd.yaml]: Perhaps add some short expiration time to them?

File: .github/workflows/cd.yaml (hunk 2 above)
Comment (ouphi) [reply .github/workflows/cd.yaml:178]: It was not removed. 
I guess it was missing because I didn't merge main yet on this branch checked out from main before the continue-on-error was merged.
But if you don't see the line deletion it means it was not removed.

File: .github/workflows/cd.yaml (hunk 4 above)
Comment (ouphi) [reply .github/workflows/cd.yaml]: This is to make terraform provider binaries executable. Because file permissions of uploaded artifacts are not maintained: https://github.com/actions/upload-artifact#permission-loss
I added a comment to explain why this is needed

File: .github/workflows/cd.yaml (hunk 3 above)
Comment (ouphi) [reply .github/workflows/cd.yaml]: For now I added `retention-days: 1` and we can implement @cpeeyush solution later.

File: .github/workflows/cd.yaml [hunk 6]
```diff
@@ -160,14 +168,32 @@ jobs:
...
+        with:
+          name: terraform-${{ matrix.participant }}
+          path: deployment/terraform
 
-      - name: 'Az CLI login'
-        uses: azure/login@v1
+      - name: 'Delete terraform resources'
+        run: |
+          : # Make terraform provider binaries executable because file permissions of uploaded artifacts are not maintained.
+          chmod -R u+x .terraform/providers
```
Comment (algattik) [inline .github/workflows/cd.yaml]: nit (just for simplicity)
```suggestion
          chmod -R +x .terraform/providers
```

File: .github/workflows/cd.yaml [hunk 7]
```diff
@@ -160,14 +168,32 @@ jobs:
...
+          # Authentication settings for Terraform AzureRM provider
+          # See https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs
+          ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+          ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+          ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+          ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+
+
+      - uses: actions/upload-artifact@v3
+        name: 'Remove terraform artifact'
```
Comment (algattik) [inline .github/workflows/cd.yaml]:
```suggestion
        name: 'Clear terraform artifact'
```

File: .github/workflows/cd.yaml [hunk 8]
```diff
@@ -160,14 +168,32 @@ jobs:
...
         with:
-          client-id: ${{ secrets.ARM_CLIENT_ID }}
-          tenant-id: ${{ secrets.ARM_TENANT_ID }}
-          subscription-id: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+          name: terraform-${{ matrix.participant }}
+          path: /dev/null
+
+
```
Comment (algattik) [inline .github/workflows/cd.yaml]:
```suggestion
```

**Review by algattik — APPROVED** [review]

**Review by cpeeyush — APPROVED** [review]
